From e7b057f883d31d594435114ad604896e306fe8ac Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Fri, 7 Jun 2019 13:57:12 +0700
Subject: [PATCH 2/4] Make tst_QOpenGLWindow pass on platforms where the
 default FBO != 0

The test would fail on Wayland with window decorations enabled because
there
window content is drawn to an FBO before it's blended with the window
decorations (also drawn by the client).

Commit-ID: 13f09acb665225a0b6bb5f20b01e01450ddb19aa
Task-number: QTBUG-51741 - detected from upstream

Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
Reviewed-by: Andy Nichols <andy.nichols@qt.io>
Reviewed-by: Paul Olav Tvete <paul.tvete@qt.io>
Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 tests/auto/gui/kernel/qopenglwindow/tst_qopenglwindow.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/auto/gui/kernel/qopenglwindow/tst_qopenglwindow.cpp b/tests/auto/gui/kernel/qopenglwindow/tst_qopenglwindow.cpp
index 1fc4967..b2e99b8 100644
--- a/tests/auto/gui/kernel/qopenglwindow/tst_qopenglwindow.cpp
+++ b/tests/auto/gui/kernel/qopenglwindow/tst_qopenglwindow.cpp
@@ -248,7 +248,7 @@ public:
 
         GLuint fbo = 0xFFFF;
         QOpenGLContext::currentContext()->functions()->glGetIntegerv(GL_FRAMEBUFFER_BINDING, (GLint *) &fbo);
-        QCOMPARE(fbo, GLuint(0));
+        QCOMPARE(fbo, QOpenGLContext::currentContext()->defaultFramebufferObject());
     }
 
     void paintGL() Q_DECL_OVERRIDE {
@@ -260,7 +260,7 @@ public:
         // Using PartialUpdateBlend so paintGL() targets a user fbo, not the default.
         GLuint fbo = 0xFFFF;
         QOpenGLContext::currentContext()->functions()->glGetIntegerv(GL_FRAMEBUFFER_BINDING, (GLint *) &fbo);
-        QVERIFY(fbo != 0);
+        QVERIFY(fbo != QOpenGLContext::currentContext()->defaultFramebufferObject());
         QCOMPARE(fbo, defaultFramebufferObject());
     }
 
@@ -272,7 +272,7 @@ public:
 
         GLuint fbo = 0xFFFF;
         QOpenGLContext::currentContext()->functions()->glGetIntegerv(GL_FRAMEBUFFER_BINDING, (GLint *) &fbo);
-        QCOMPARE(fbo, GLuint(0));
+        QCOMPARE(fbo, QOpenGLContext::currentContext()->defaultFramebufferObject());
     }
 };
 
-- 
2.7.4

